packages/flutter_tools/test/general.shard/android/native_assets_test.dart

```diff
@@ -0,0 +1,386 @@
+              path: AssetAbsolutePath(Uri.file('libbar.so')),
+            ),
+            Asset(
+              id: 'package:bar/bar.dart',
+              linkMode: LinkMode.dynamic,
+              target: native_assets_cli.Target.macOSX64,
+              path: AssetAbsolutePath(Uri.file('libbar.so')),
+            ),
+          ],
+        ),
+      ),
+    );
+    expect(
+      (globals.logger as BufferLogger).traceText,
+      stringContainsInOrder(<String>[
+        'Dry running native assets for android.',
+        'Dry running native assets for android done.',
+      ]),
+    );
+    expect(
+      nativeAssetsYaml,
+      projectUri.resolve('build/native_assets/android/native_assets.yaml'),
+    );
+    expect(
+      await fileSystem.file(nativeAssetsYaml).readAsString(),
+      contains('package:bar/bar.dart'),
+    );
+  });
+
+  testUsingContext('build with assets but not enabled', () async {
+    final File packageConfig = environment.projectDir.childFile('.dart_tool/package_config.json');
+    await packageConfig.parent.create();
+    await packageConfig.create();
+    expect(
+      () => buildNativeAssetsAndroid(
+        androidArchs: <AndroidArch>[AndroidArch.arm64_v8a],
+        targetAndroidNdkApi: 21,
+        projectUri: projectUri,
+        buildMode: BuildMode.debug,
+        fileSystem: fileSystem,
+        yamlParentDirectory: environment.buildDir.uri,
+        buildRunner: FakeNativeAssetsBuildRunner(
+          packagesWithNativeAssetsResult: <Package>[
+            Package('bar', projectUri),
+          ],
+        ),
+      ),
+      throwsToolExit(
+        message: 'Package(s) bar require the native assets feature to be enabled. '
+            'Enable using `flutter config --enable-native-assets`.',
+      ),
+    );
+  });
+
+  testUsingContext('build no assets', overrides: <Type, Generator>{
+    FeatureFlags: () => TestFeatureFlags(isNativeAssetsEnabled: true),
+    ProcessManager: () => FakeProcessManager.empty(),
+  }, () async {
+    final File packageConfig = environment.projectDir.childFile('.dart_tool/package_config.json');
+    await packageConfig.parent.create();
+    await packageConfig.create();
+    await buildNativeAssetsAndroid(
+      androidArchs: <AndroidArch>[AndroidArch.arm64_v8a],
+      targetAndroidNdkApi: 21,
+      projectUri: projectUri,
+      buildMode: BuildMode.debug,
+      fileSystem: fileSystem,
+      yamlParentDirectory: environment.buildDir.uri,
+      buildRunner: FakeNativeAssetsBuildRunner(
+        packagesWithNativeAssetsResult: <Package>[
+          Package('bar', projectUri),
+        ],
+      ),
+    );
+    expect(
+      environment.buildDir.childFile('native_assets.yaml'),
+      exists,
+    );
+  });
+
+  testUsingContext('build with assets', overrides: <Type, Generator>{
+    FeatureFlags: () => TestFeatureFlags(isNativeAssetsEnabled: true),
+    ProcessManager: () => FakeProcessManager.empty(),
+  }, () async {
+    if (const LocalPlatform().isWindows) {
```

Comment:
This pattern of having a test pass on windows by returning out before the code executes feels like a foot gun? Is there not another way to filter what tests run on which hosts?

Reply:
Replaced with a skip as found in other places in the code.